staff/gianluca-bragaglia/pre-course/bingo-pro-html/bingo-pro-HTML.js

```diff
@@ -0,0 +1,329 @@
+
+
+    }
+    generateArrRandom();
+
+
+
+    function generateRandomNum() {
+
+        var numRandom = arrRandomNum[arrRandomNum.length - 1];
+        var r = arrRandomNum.length;
+        return numRandom;
+
+    }
+
+
+
+
+    function newTurn() {
+
+        var rand = generateRandomNum();
+        var line = ['X', 'X', 'X', 'X', 'X'];
+        playerPoints--;
+
+        matchN.style.display = "none";
+        cardCont.style.display = "inline-block";
+        document.getElementById('rand').innerHTML = rand;
+        pointsN.innerHTML = (playerName + ' Your points are: ' + playerPoints);
+
+
+        function newNumber() {
+
+            rand = generateRandomNum();
+            playerPoints--;
+
+        }
+
+
+
+        btnNewN.click = function() {
+
+            arrRandomNum.pop();
+            newNumber();
+
+        }
+
+
+
+        for (var i = 0; i < 5; i++) {
+
+            if (card15[i] === rand) {
+
+                card15.splice(i, 1, 'X');
+                card15N.innerHTML = card15.join("&nbsp;&nbsp;&nbsp;");
+                matchN.style.display = "inline-block";
+                matchN.innerHTML = ('Good!! Match number: ' + rand);
+
+            }
+        }
+
+        for (var i = 0; i < 5; i++) {
+
+            if (card510[i] === rand) {
+
+                card510.splice(i, 1, 'X');
+                card510N.innerHTML = card510.join("&nbsp;&nbsp;&nbsp;");
+                matchN.style.display = "inline-block";
+                matchN.innerHTML = ('Good!! Match number: ' + rand);
+            }
+        }
+
+        for (var i = 0; i < 5; i++) {
+
+            if (card1015[i] === rand) {
+
+                card1015.splice(i, 1, 'X');
+                card1015N.innerHTML = card1015.join("&nbsp;&nbsp;&nbsp;");
+                matchN.style.display = "inline-block";
+                matchN.innerHTML = ('Good!! Match number: ' + rand);
+
+            }
+        }
+
+
+        arrRandomNum.pop();
+
+        function areArrsMatch(arr1, arr2) {
+            return arr1.sort().toString() === arr2.sort().toString()
+        }
+
+        if (areArrsMatch(line, card15) && countA !== 0) {
+            countA = 0;
+            if (countB == 0 && countC == 0) {
+
+                matchN.innerHTML = 'BINGO!!!';
+                status = 'win';
+                pointsN.innerHTML = (playerName + ' Your points are: ' + playerPoints);
+                buttonsN.style.display = "none";
+                randDiv.style.display = "none";
+                cardCont.style.display = "none";
+                pointsN.style.display = "none";
+                arrPlayers.push({ name: playerName, puntos: playerPoints, status: status });
+                results();
+            } else {
+                matchN.innerHTML = 'LINEA!!!';
+            }
+
+
+        } else if (areArrsMatch(line, card510) && countB !== 0) {
+            countB = 0;
+            if (countA == 0 && countC == 0) {
```

Comment:
Maybe would be more accurated to use === instead of == .

[Discussion post- StackOverFlow](https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons)

Reply:
Thanks, I have already fixed